specification/basic_concepts/basic_concepts.tex

```diff
@@ -1,33 +1,62 @@
 migration path if backward-incompatible changes are introduced.
-Some data structures are standard and defined by the protocol specification (of which only a small 
+Some data types are standard and defined by the protocol specification (of which only a small
 subset are required); others may be specific to a particular application or vendor.
 
-Since every message or service data type has its own unique data type ID,
-and each node in the network has its own unique node ID,
-a pair of data type ID and node ID can be used to support redundant nodes with identical
-functionality inside the same network.
-
-Message and service data structures are defined using the Data Structure Description Language
-(DSDL) (chapter \ref{sec:dsdl}).
-A DSDL description can be used to automatically generate the serialization/deserialization code
-for every defined data structure in a particular programming language.
+Message exchanges between nodes are grouped into \emph{subjects} by their semantic meaning.
+Message exchanges belonging to the same subject use same message data type down to the major version
+(minor versions are allowed to differ) and pertain to same function or process within the system.
+
+Request/response exchanges between nodes are grouped into \emph{services} by their semantic meaning,
+like messages are grouped into subjects.
+Requests and their corresponding responses that belong to the same service use same service data type down to
+the major version (minor versions are allowed to differ; as a consequence, the minor data type version number
+of a service response may differ from that of its corresponding request) and pertain to same function.
+
+Each message subject is identified by a unique natural number -- a \emph{subject ID};
+likewise, each service is identified by a unique \emph{service ID}.
+An umbrella term \emph{port ID} is used to refer either to a subject ID or to a service ID
+(port identifiers have no direct manifestation in the construction of the protocol,
+but they are convenient for discussion).
+
+Message and service data types\footnote{Data types are the core concept of UAVCAN.}
+are defined using the \emph{data structure description language} (DSDL) (chapter \ref{sec:dsdl}).
+A DSDL definition specifies the name, major version, minor version, the data structure,
+and an optional predefined port ID of the data type.
+Message data types always contain exactly one data structure, whereas
+service data types contain two data structures: one for request, and the other for response.
+
+If a port identifier is defined by a data type definition, such identifier is said to be a
+\emph{regulated port identifier}
+(by inheritance, the same naming principle applies to \emph{regulated subject ID} and \emph{regulated service ID}).
+If a regulated port ID is not provided with a data type definition,
+it can be assigned at runtime on a per-application basis,
+in which case it is said to be an \emph{unregulated port identifier}.
```

Comment:
"Standard" is probably a poor fit because we also use this word to describe specification-defined data types ("standard data types"). In this discussion, we have agreed to use separate ranges for regulated and application-specific identifiers, hence the split you mentioned: https://forum.uavcan.org/t/on-subject-service-id-ranges/195/4?u=pavel.kirienko

Reply:
How you define regulated in the text is not how it's used in that discussion. See my previous comment.